src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/ImportSectionNode.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace ILCompiler.DependencyAnalysis.ReadyToRun
+{
+    public class ImportSectionNode : EmbeddedObjectNode
+    {
+        private ArrayOfEmbeddedDataNode<Import> _imports;
+        private ArrayOfEmbeddedPointersNode<Signature> _signatures;
+        private readonly CorCompileImportType _type;
+        private readonly CorCompileImportFlags _flags;
+        private readonly byte _entrySize;
+
+        public ImportSectionNode(CorCompileImportType importType, CorCompileImportFlags flags, byte entrySize)
+        {
+            _type = importType;
+            _flags = flags;
+            _entrySize = entrySize;
+
+            _imports = new ArrayOfEmbeddedDataNode<Import>($"imports_{NodeIdentifier}_start", $"imports_{NodeIdentifier}_end", null);
+            _signatures = new ArrayOfEmbeddedPointersNode<Signature>($"signaures_{NodeIdentifier}_start", $"signatures_{NodeIdentifier}_end", null);
+        }
+        
+        private string NodeIdentifier => $"_{_type}_{_flags}_{_entrySize}";
+
+        public void AddImport(NodeFactory factory, Import import)
+        {
+            _imports.AddEmbeddedObject(import);
+
+            var signature = import.GetSignature(factory);
+            if (signature != null)
+            {
+                _signatures.AddEmbeddedObject(signature);
```

Comment:
More of a question than anything else: is the import and import's signature related? How is the runtime going to match them up if the entry in the signature array can be missing?

Reply:
Some imports types have signatures and some don't. A ImportSectionNode contains imports of the same type so they'll all have signatures or all not have them. An assert for consistency would probably be useful here